datalake/src/main/java/com/sequenceiq/datalake/flow/create/handler/EnvWaitHandler.java

```diff
@@ -56,25 +77,65 @@ protected Selectable doAccept(HandlerEvent<EnvWaitRequest> event) {
+
+            // TODO Extract this logic into a separate flow step - CB-17514
+            sdxClusterRepository.findById(datalakeId)
+                    .ifPresentOrElse(this::scheduleStorageConsumptionCollectionIfNeeded,
+                            () -> {
+                                throw notFound("SDX cluster", datalakeId).get();
+                            });
+
             response = new EnvWaitSuccessEvent(datalakeId, userId, detailedEnvironmentResponse);
             setEnvCreatedStatus(datalakeId);
         } catch (UserBreakException userBreakException) {
             LOGGER.error("Env polling exited before timeout. Cause: ", userBreakException);
             response = new SdxCreateFailedEvent(datalakeId, userId, userBreakException);
         } catch (PollerStoppedException pollerStoppedException) {
-            LOGGER.error("Env poller stopped for sdx: {}", datalakeId, pollerStoppedException);
+            LOGGER.error("Env poller stopped for SDX: {}", datalakeId, pollerStoppedException);
             response = new SdxCreateFailedEvent(datalakeId, userId,
                     new PollerStoppedException("Env wait timed out after " + durationInMinutes + " minutes"));
         } catch (PollerException exception) {
-            LOGGER.error("Env polling failed for sdx: {}", datalakeId, exception);
+            LOGGER.error("Env polling failed for SDX: {}", datalakeId, exception);
             response = new SdxCreateFailedEvent(datalakeId, userId, exception);
         } catch (Exception anotherException) {
-            LOGGER.error("Something wrong happened in sdx creation wait phase", anotherException);
+            LOGGER.error("Something wrong happened in SDX creation wait phase", anotherException);
             response = new SdxCreateFailedEvent(datalakeId, userId, anotherException);
         }
         return response;
     }
 
+    private void scheduleStorageConsumptionCollectionIfNeeded(SdxCluster sdxCluster) {
+        String accountId = sdxCluster.getAccountId();
+        if (consumptionEnabled && entitlementService.isCdpSaasEnabled(accountId)) {
+            scheduleStorageConsumptionCollection(sdxCluster);
+        } else {
+            LOGGER.info("Skipping storage consumption collection scheduling because " +
+                    (consumptionEnabled ? String.format("CDP_SAAS entitlement is missing for account '%s'", accountId) : "it is disabled for the deployment"));
+        }
+    }
+
+    private void scheduleStorageConsumptionCollection(SdxCluster sdxCluster) {
+        String storageBaseLocation = sdxCluster.getCloudStorageBaseLocation();
+        if (!Strings.isNullOrEmpty(storageBaseLocation)) {
+            scheduleStorageConsumptionCollectionForStorageLocation(sdxCluster, storageBaseLocation);
+        } else {
+            LOGGER.warn("Skipping storage consumption collection scheduling for storage base location because it is not provided: " +
```

Comment:
question point: this case might not even exist

Reply:
I left this mainly for sake of `MOCK` and `YARN` platforms, in case the logic would be ever executed when using either of them.

Reply:
Note that the above logic has been moved over to `ConsumptionService`.